core/lib/src/response/named_file.rs

```diff
@@ -80,13 +96,38 @@ impl NamedFile {
 /// implied by its extension, use a [`File`] directly.
 impl Responder<'_> for NamedFile {
     fn respond_to(self, req: &Request<'_>) -> response::Result<'static> {
-        let mut response = self.1.respond_to(req)?;
-        if let Some(ext) = self.0.extension() {
+        if let Some(if_modified_since) = req.headers().get("If-Modified-Since").next() {
+            if let Some(last_modified) = &self.modified {
+                let if_modified_since = match httpdate::parse_http_date(if_modified_since) {
+                    Ok(if_modified_since) => if_modified_since,
+                    Err(_err) => return Response::build().status(Status::BadRequest).ok(),
+                };
+
+                if last_modified
+                    .duration_since(if_modified_since)
+                    .unwrap_or(Duration::from_secs(60))
+                    <= Duration::from_secs(1)
```

Comment:
Can this be a simpler comparison, like `if last_modified <= if_modified_since`? My guess is it can be, as long as both dates are `HttpDate`s.